lib/paper_trail/events/destroy.rb

```diff
@@ -21,6 +21,13 @@ def data
         if record_object?
           data[:object] = recordable_object(false)
         end
+        if record_object_changes?
+          # Rails' implementation returns nothing on destroy :/
+          changes = @record.attributes.map { |attr, value| [attr, [value, nil]] }.to_h
+          changes = serialize_changes(changes)
+          changes = recordable_object_changes(changes)
+          data[:object_changes] = changes
+        end
```

Comment:
This method is getting a bit long. I think we can extract some part of this into a new method. Maybe something like

```ruby
if record_object_changes?
  data[:object_changes] = object_changes
end
# ..

private

# Rails' implementation returns nothing on destroy, so we build our own hash.
def object_changes
  changes = @record.attributes.map { |attr, value| [attr, [value, nil]] }.to_h
  changes = serialize_changes(changes)
  recordable_object_changes(changes)
end
```

What do you think?

Reply:
The previous level of abstraction we had in `changes` clearly wasn't right, since both the update and destroy events would customize the underlying changeset.

Currently every event has these two lines in common w/out any modifications:
```ruby
changes = serialize_changes(changes)
changes = recordable_object_changes(changes)
```

What if we introduced a method to wrap those?

```ruby
# destroy.rb
if record_object_changes?
  changes = @record.attributes.map { |attr, value| [attr, [value, nil]] }.to_h
  data[:object_changes] = prepare_object_changes(changes)
end

# base.rb
def prepare_object_changes(changes)
  recordable_object_changes(serialize_changes(changes))
end